File: src/main/java/com/patres/kotlinplusplus/d_classes/java/ClassesApp.java
package com.patres.kotlinplusplus.d_classes.java;

public class ClassesApp {

    public static void main(String[] args) {
        Person person1 = new Person("Tom", "Apple");
        Person person2 = new Person("Tom", "Apple");

        System.out.println(person1.equals(person2));
        System.out.println(person1 == person2);

        String name = person1.getName();
        String surname = person1.getSurname();
        System.out.println(name + " " + surname);

        Property property = new Property();
        property.setValue("New value");

        System.out.println(Property.VERSION);
    }
    
}

File: src/main/java/com/patres/kotlinplusplus/d_classes/kotlin/ClassesApp.kt
package com.patres.kotlinplusplus.d_classes.kotlin

import com.patres.kotlinplusplus.d_classes.java.Property


fun main() {
    val person = Person("Tom", "Apple")
    val (name, surname) = person // componentN()
    person.equals(person)
    person.hashCode()
    person.copy()
    person.toString() // Person(name=Tom, surname=Apple)

    val person1 = Person("Tom", "Apple")
    val person2 = Person("Tom", "Apple")

    println(person1 == person2)
    println(person1 === person2)


    println("$name $surname")

    val property = Property()
    property.value = "New value"
    property.key = "New key"


    println(Property.VERSION)


}

